### PR TITLE
Data: Bug fixes & 100 % code coverage with @covers

### PR DESCRIPTION
**Pull request by lukasbestle**
## Describe the PR

I have refactored and tested the `Data` package:

- Improve doc blocks
- Smaller code improvements
- Fixes and improvements to the `Txt` handler:
  - Multiple accidental field dividers in a row (in lines following each other) were not escaped, breaking the output
  - Those escaped field dividers are now also unescaped again when decoding to improve the editing experience (the whole escaping business now only happens behind the scenes)
  - Cosmetic fix: Fields with multiple lines now won't have a space at the end of the key's line
- 100 % code coverage with unit tests (from 95 %), including all edge cases I could think of (therefore the fixes above)

## Todos
- [x] Add unit tests for fixed bug/feature
- [x] Pass all unit tests
- [x] Fix code style issues with CS fixer and `composer fix`
- [x] If needeed, in-code documentation (DocBlocks etc.)

### CODE REVIEW

File: src/Data/Data.php [hunk 1]
```diff
@@ -41,40 +41,40 @@ class Data
      * @var array
      */
     public static $handlers = [
-        'json' => 'Kirby\Data\Json',
-        'php'  => 'Kirby\Data\PHP',
-        'txt'  => 'Kirby\Data\Txt',
-        'yaml' => 'Kirby\Data\Yaml',
+        'json' => Json::class,
```
Comment (bastianallgeier) [inline src/Data/Data.php]: There was a reason for the strings. They don't trigger the autoloader. With the class constant the autoloader will automatically load the classes even if they are not used afterwards.

**Review by bastianallgeier — CHANGES_REQUESTED** [review]
I think we should move back to the string class names. Did you also check if the new regex has any negative impacts on performance?

File: src/Data/Data.php (hunk 1 above)
Comment (lukasbestle) [reply src/Data/Data.php]: Oh wow, I didn't know that. That's definitely stupid then. I have reverted this change.

**Comment by lukasbestle** [comment]
The new regex is actually faster because it needs to do less replacing stuff and only matches at all if the lookbehind assertion at the beginning matches.

I cobbled together a small comparison test:

```php
<?php

$text = '
Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

----

Test

----

Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

----

Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

----

Another test

----

The end';

function test($regex, $replace, $text) {
    $start = microtime(true);
    for ($i = 0; $i < 100000; $i++) {
        preg_replace($regex, $replace, $text);
    }
    return microtime(true) - $start . "\n";
}

echo 'Old:  ' . test('!(\n|^)----(.*?\R*)!', '$1\\----$2', $text);
echo 'New1: ' . test('!(?<=\n|^)----(?=.*\n|.*$)!', '\\----', $text);
echo 'New2: ' . test('!(?<=\n|^)----!', '\\----', $text);
```

The `New1` regex is the original one from this PR. I have now optimized it even further.

Results on my machine:

```
Old:  0.29804396629333
New1: 0.27344298362732
New2: 0.27114510536194
```

**Review by bastianallgeier — APPROVED** [review]
